### PR TITLE
Add roundType ID to `Round` output under `events`

### PR DESCRIPTION
**Pull request by gregorbg**
Bonus: Fixes deprecated URLs pointing to the old `WcaOnRails` folder of the monolith

### CODE REVIEW

File: specification.md [hunk 1]
```diff
@@ -353,7 +353,8 @@ Represents data of a round held at the competition.
 | Attribute | Type | Description |
 | --- | --- | --- |
 | `id` | `String` | The round identifier of the form `{eventId}-r{roundNumber}`. *Note: this is a valid [`ActivityCode`](#activitycode).* |
-| `format` | `"1"\|"2"\|"3"\|"a"\|"m"` | The round format. Look [here](https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/db/seeds/formats.seeds.rb) for the list of all the WCA formats. |
+| `format` | `"1"\|"2"\|"3"\|"a"\|"m"` | The round format. Look [here](https://github.com/thewca/worldcubeassociation.org/blob/master/db/seeds/formats.seeds.rb) for the list of all the WCA formats. |
+| `roundType` | `"0"\|"1"\|"2"\|"3"\|"b"\|"c"\|"d"\|"e"\|"f"\|"g"\|"h"` | The round type. Look [here](https://github.com/thewca/worldcubeassociation.org/blob/master/db/seeds/round_types.seeds.rb) for the list of all the WCA round types. (In essence, a Round Type describes this round in relation to other rounds of this competition.) |
```
Comment (timreyn) [inline specification.md:357]: It seems a little redundant to expose the "combined" roundTypes (c,d,e,g,h), since they're rendered the same as f,1,2,3, and q. Maybe we should not include these in the API?

(Personally I'd love to get rid of those roundTypes entirely, but that's a larger discussion)

Comment (jonatanklosko) [reply specification.md:357]: I'd guess the reason the round type hasn't been included so far is that it can be determined based on round number and whether it has a cutoff, so this information is somewhat redundant. I don't have strong opinion though, if there's a good use case for including the summarised field, that's ok.

Comment (FinnIckler) [reply specification.md:357]: It's because of this PR https://github.com/thewca/worldcubeassociation.org/pull/9262. We use wcif to render the events tab and translating the activity code to a i18n string is really annoying and would involve a lot of `if roundNumber == x && maxRounds == y return z ` with this you can just do `i18n.t(rounds.${round.roundType}.cellName)`

Comment (jonatanklosko) [reply specification.md:357]: You could have a single function that takes the relevant info and generates the round type id, to encapsulate all the logic in a single place, then invoke it in the interpolation.

Comment (FinnIckler) [reply specification.md:357]: sure, but why go through the trouble of converting to wcif and then "unconverting" the wcif back to the round type by duplicating code from the ruby side into js if you can include that data in the output (either through the wcif or as additional data to the react component if this is a controversial change)

Comment (jonatanklosko) [reply specification.md:357]: It's mostly about avoiding duplication, but again, I don't have strong opinion :)

**Comment by FinnIckler** [comment]
We are choosing to create new frontend logic instead, thanks for the comments

**Comment by gregorbg** [comment]
I want to open this discussion again. Deducing the activity code "on the fly" in the frontend causes a lot of inconveniences.

Essentially, every bit of code that needs to localize an activity code also needs extended information about the WCIF Round (to see whether it has a cutoff) _and_ the WCIF Event (to see how many rounds there are in total). So the signature of `function localizeActivityCode(activityCode)` becomes `function localizeActivityCode(activityCode, wcifRound, wcifEvent)`.

In other words, you have to pass around almost the entire WCIF every time you want to localize **one** AC. This is most obvious in the EditSchedule and ShowSchedule pages, but it also touches on other, less trivial parts of the codebase. See for example the complications necessary to "properly" (i.e. human-readable, localized) display cumulative events in the ShowEvents tab: https://github.com/thewca/worldcubeassociation.org/pull/9365

TL;DR: Yes, it is redundant information. But deducing it on-the-fly is painful and also somewhat redundant (passing around the entire list of WCIF Events including all their rounds, _just in case there might be a cumulative time limit somewhere_)

**Comment by jonatanklosko** [comment]
@gregorbg your call!
